### ISSUE TITLE
[2.0.10] Bug - ODM - findByld

### ISSUE THREAD

**Issue by Mo121Montazeri** [phalcon/cphalcon#12010]
Hi,
In ODM models, if id doesn't found in Database when using findByld function, the false value should be returned.

```php
$id = 'not exist id';
$user = Users::findById($id);

if($user == false){
      // redirect
}
```

**Comment by sergeyklay** [comment]
What about:
```php
if (!$user->count()) {
    // redirect
}
```

**Comment by Mo121Montazeri** [comment]
when call findById($id) with not exist id error

`
Invalid object ID
#0 [internal function]: MongoId->__construct('s')
#1 [internal function]: Phalcon\Mvc\Collection::findById('s')
#2 /Users/iMohammad/Development/Phalcon/apps/backend/controllers/AdminAbstract.php(128): call_user_func(Array, 's')
#3 [internal function]: App\Backend\Controllers\AdminAbstract->confirmAction('s')
#4 [internal function]: Phalcon\Dispatcher->dispatch()
#5 /Users/iMohammad/Development/Phalcon/public/index.php(44): Phalcon\Mvc\Application->handle()
#6 {main}`

**Comment by sergeyklay** [comment]
Hmm.. Okay. What about:
```php
$user = Users::findById(\MongoId('545eb081631d16153a293a66'));
```

In any case, what do you suggest? To add regexp to the `findById` like this?
```js
/^(?=[a-f\d]{24}$)(\d+[a-f]|[a-f]+\d)/i
```

**Comment by Mo121Montazeri** [comment]
Whenever this error occurs, it means id characters count are not 24.
I did not pay attention to that

**Comment by Mo121Montazeri** [comment]
yes, it's good for findById
tnx

**Comment by sergeyklay** [comment]
Fixed in the `2.1.x` branch.
